src/hotspot/share/gc/serial/tenuredGeneration.hpp

```diff
@@ -168,7 +168,7 @@ class TenuredGeneration: public Generation {
   //
   // The "obj_size" argument is just obj->size(), passed along so the caller can
   // avoid repeating the virtual call to retrieve it.
-  oop promote(oop obj, size_t obj_size);
+  oop allocate_for_promotion(oop obj, size_t obj_size);
```

Comment:
The first paragraph above needs to be adjusted to the new decreased functionality.